Elastic vector search and chunk

hadonas opened the pull request:
PR Check List

[ ] Self Code Review fininshed
[ ] Formatting fininshed
[ ] Lint/Unit/Module Test fiinished

What does this PR do

카테고리 시스템을 업데이트하여 3단계가 아닌 다양한 레벨로 생성할 수 있습니다 (삭제/변경은 안됨)
카테고리 생성시 인공지능이 카테고리의 디테일을 생성하게 할 수 있습니다.
elasticsearch를 도입해 카테고리와 문서를 elasticsearch의 샤드들에 청킹하여 입력하고, shard-aware vector search로 검색하게 했습니다. 이에 따라 /chat 페이지의 llm 응답시간이 약 30초에서 약 10초로 줄었습니다.

테스트방법

env.example을 .env로 복사하고, .env의 LLM 설정(ex. api key)을 마친 뒤, 루트 디렉터리에서 docker-compose up -d --build를 입력합니다.
이후 localhost:3000에서 프론트엔드를 볼 수 있습니다.

admin 계정은 email이 [admin@company.com](mailto:admin@company.com)이고 password가 admin123입니다.
참고: xlsx 파일은 아직 지원하지 않습니다. docx파일을 위주로 테스트해주세요.

테스트화면

![4329B703-903E-4308-8DD7-9DC7E91020E7_1_105_c](https://github.com/user-attachments/assets/c37df954-7065-4a1f-818b-0178fddad127)
![42B2A9CD-E0CC-4D78-9FA2-BA75837C1AB6_1_105_c](https://github.com/user-attachments/assets/cc2bc8f9-3e93-48a6-b8bb-cfc31d13dfdd)

hadonas committed:
카테고리 추가 시스템 및 분류 rag, 재분류 시스템 등 업데이트

hadonas committed:
elastic rrf hybrid search and chunking

hadonas committed:
changed threshold to .2

hadonas committed:
elasticsearch chunking and sharding

hadonas committed:
deleted unused search algorithms and minor fixes

hadonas committed:
updated documentation

hadonas committed:
text change

hadonas committed:
utilized hierarchy system more

hadonas committed:
minor bugfixes

hadonas committed:
deletion 디버깅

J-nowcow commented:
bugbot run

J-nowcow commented:
> bugbot run

다른사람꺼엔 안되나 보네요 까비

J-nowcow commented:
수고하셨습니다~
llm 응답시간이 감소했다는 것을 증명할 수 있는 장표가 있으면 좋을 것 같습니다.
예를 들어 사전에 여러 종류의 input을 준비해놓고, 각 input에 대해 pr 이전 / 이후 버전에서의 속도 차이를 측정한 다음 간단한 표나 그래프로 표현하는 방식이 있을 것 같습니다.

hadonas commented:
> 수고하셨습니다~ llm 응답시간이 감소했다는 것을 증명할 수 있는 장표가 있으면 좋을 것 같습니다. 예를 들어 사전에 여러 종류의 input을 준비해놓고, 각 input에 대해 pr 이전 / 이후 버전에서의 속도 차이를 측정한 다음 간단한 표나 그래프로 표현하는 방식이 있을 것 같습니다.

감사합니다. 전반적으로 테스트케이스들을 만들어봐야 할 것 같네요

younyoungieo commented on token.txt:
```diff
@@ -0,0 +1 @@
+eyJhbGciOiJIUzI1NiJ9.eyJyb2xlcyI6WyJBRE1JTiIsIkFQUFJPVkVSIiwiVVNFUiJdLCJ1c2VySWQiOiI2OGMyMzEwYmI3ZmNkMjJhZjkxODEwNzkiLCJzdWIiOiJhZG1pbkB0ZXN0LmNvbSIsImlzcyI6Im1kbS5sb2NhbCIsImlhdCI6MTc1ODYwMzkyMiwiZXhwIjoxNzU4NjkwMzIyfQ.dnbWNDXpu8gLgdFPBfihSkdqoxM0n5WHGvQK6XTsu5k
```
이거 다 진짜 토큰이에요? ㄷㄷ

kyleee0522 commented on README.md:
```diff
@@ -129,23 +129,22 @@ npm run build
+3) RRF 결합 후 부족 시 전체 샤드 확장
+4) fallback match 보강
+
+카테고리 분류 후보도 동일한 하이브리드 로직(`CLASSIFICATION_CANDIDATE_COUNT * 2`)을 거쳐 상위 5개(기본값)를 반환합니다.
```
고생하쎳습니다.

hadonas replied:
감사합니다

HyunjunGil reviewed (approved):
LGTM

hadonas committed:
문서작업

feat: Add GitHub Actions CI workflow with Trivy security scanning

- Add comprehensive CI pipeline with security scan, backend build, and frontend build jobs
- Include Trivy vulnerability scanner with SARIF upload to GitHub Security tab
- Add Java 17 and Node.js 18 build environments
- Include caching for Gradle and npm dependencies
- Run tests for both backend and frontend

fix: Update CodeQL action from v2 to v3

- Update github/codeql-action/upload-sarif from v2 to v3
- Change input parameter from sarif_file to sarif-file (v3 syntax)
- Resolves deprecation warning for v2

fix: Fix SARIF file path for CodeQL upload

- Add explicit relative path prefix (./) to sarif-file parameter
- Add debugging step to check SARIF file existence and location
- Ensures CodeQL action can find the trivy-results.sarif file

Update ci.yml

feat: Add GitHub Actions CI workflow with Trivy security scanning

- Add comprehensive CI pipeline with security scan, backend build, and frontend build jobs
- Include Trivy vulnerability scanner with SARIF upload to GitHub Security tab
- Add Java 17 and Node.js 18 build environments
- Include caching for Gradle and npm dependencies
- Run tests for both backend and frontend

fix: Update CodeQL action from v2 to v3

- Update github/codeql-action/upload-sarif from v2 to v3
- Change input parameter from sarif_file to sarif-file (v3 syntax)
- Resolves deprecation warning for v2

fix: Fix SARIF file path for CodeQL upload

- Add explicit relative path prefix (./) to sarif-file parameter
- Add debugging step to check SARIF file existence and location
- Ensures CodeQL action can find the trivy-results.sarif file

fix: Correct CodeQL upload-sarif parameter name

- Change sarif-file back to sarif_file (underscore not hyphen)
- CodeQL action v3 still uses underscore naming convention
- Fixes 'Unexpected input sarif-file' error

문서작업

문서작업
